analytic_engine/src/sst/parquet/meta_data.rs

```diff
@@ -120,14 +125,39 @@ pub struct RowGroupFilterBuilder {
 }
 
 impl RowGroupFilterBuilder {
-    pub(crate) fn with_num_columns(num_col: usize) -> Self {
-        Self {
-            builders: vec![None; num_col],
-        }
+    pub(crate) fn new(record_schema: &RecordSchemaWithKey) -> Self {
+        let key_idx = record_schema.primary_key_idx();
+        let builders = record_schema
+            .columns()
+            .iter()
+            .enumerate()
+            .map(|(i, col)| {
+                if key_idx.contains(&i) {
+                    return None;
+                }
```

Comment:
Drop L129, then update this to:
```suggestion
                if record_schema.is_primary_key_index(i) {
                    return None;
                }
```